process/coordinator/process.go

```diff
@@ -1401,19 +1433,26 @@ func (tc *transactionCoordinator) getAllTransactions(body *block.Body) map[block
 }
 
 func (tc *transactionCoordinator) verifyGasLimit(
+	header data.HeaderHandler,
 	body *block.Body,
 	mapMiniBlockTypeAllTxs map[block.Type]map[string]data.TransactionHandler,
 ) error {
-	for _, miniBlock := range body.MiniBlocks {
+	for index, miniBlock := range body.MiniBlocks {
 		isCrossShardMiniBlockFromMe := miniBlock.SenderShardID == tc.shardCoordinator.SelfId() &&
 			miniBlock.ReceiverShardID != tc.shardCoordinator.SelfId()
 		if !isCrossShardMiniBlockFromMe {
 			continue
 		}
-
 		if miniBlock.Type == block.SmartContractResultBlock {
 			continue
 		}
+		if tc.flagScheduledMiniBlocks.IsSet() {
+			miniBlockHeader := header.GetMiniBlockHeaderHandlers()[index]
```

Comment:
here as well, can we add a protection for index out of bounds?
I know it cannot happen in the current setup, but it would be safer if we check it.

Reply:
Done